index.js

```diff
@@ -40,19 +46,36 @@ if (argv.dictionary) {
    * Humans do mistakes, so we allow several attempts
    * to enter the code.
    */
-  var attemptCounts = 0;
-  while (true) {
+  let attemptCounts = 0;
+  /**
+   * Token will expire when token timer reaches
+   * the max alloted time, in seconds.
+   */
+  let tokenTimer = 0;
+  /**
+   * Creates an object which calls checkFunction once
+   * a second. Is terminated with clearInterval(timerInterval);
+   */
+  const timerInterval = setInterval(checkToken, 1000);
```

Comment:
I know that in the ticket I only mentioned that `setTimeout` shouldn't be used, but `setInterval` is the same idea here. I haven't try that but just judging from what I can see, this will call `checkToken` every second regardless of whether user entered an input or not, which is not a good behavior.

Instead, I suggest to do it without using of any sort of timer function and instead compare timestamp of when token was generated to the timestamp of when user attempted to guess the token.